app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -43,7 +44,15 @@ class IntentReceiverActivity : Activity() {
 
     fun processIntent(intent: Intent) {
         // Call process for side effects, short on the first that returns true
-        val processor = getIntentProcessors().firstOrNull { it.process(intent) }
+        val private = settings().openLinksInAPrivateTab
+        intent.putExtra(PRIVATE_BROWSING_MODE, private)
+        if (private) {
+            components.analytics.metrics.track(Event.OpenedLink(Event.OpenedLink.Mode.PRIVATE))
+        } else {
+            components.analytics.metrics.track(Event.OpenedLink(Event.OpenedLink.Mode.NORMAL))
+        }
+
+        val processor = getIntentProcessors(private).firstOrNull { it.process(intent) }
```

Comment:
So AFAICT this `getIntentProcessors` will also process `externalAppIntentProcessors` and others. Before even though it wasn't working (lol) it was attempting to just add this extra on the custom tab and intent processors. Will there be any bad side effects from adding this extra to other intent entry points?

Reply:
There shouldn't be side effects when we add extra intents. If we want to be super safe we can set the boolean on both `this.intent` (before) and the home activity intent (now).

Reply:
Sorry that was a confusing comment. I meant as an example - now the TWA/PWA externalAppIntentProcessors will also be attaching `PRIVATE_BROWSING_MODE`, right? Does that mean PWAs will also be attempting to present / be in PBM?

Reply:
Honestly we wanted it to work that way in the first place, so if they do present in PBM that fixes another bug :)

We can do another pass in the future to re-evaulate what data is getting passed in intents like this.